docs/specs/lcb-packages.md

```diff
@@ -0,0 +1,246 @@
+            iossimulator-i386-4_3/
+                <compiled foreign code for 32-bit iOS Simulator min version 4.3>
+            windows-i386/
+                <compiled foreign code for 32-bit windows>
+            <arch>/
+                ...
+        resources/
+            <resource files for compiled modules>
+        support/
+            <structure defined by loading environment>
+        docs/
+            <structure defined by loading environment>
+      
+The support, docs and manifest file are not used by the LCB runtime itself and
+provide supporting information for the environment that is intended to load and
+use the package.
+
+Note: At present a package only ever contains a single compiled module which is
+placed as <root>/module.lcm. The revised package format will see all modules
+within the modules subfolder.
+
+## Contents
+
+### Mimetype (pending)
+
+This file has a single line of text which gives the MIME type for the document.
+It must be the first file included in the zip archive and be stored in an
+uncompressed form. This ensures that 'magic number' mechanisms can be used to
+determine the actual type of the package file.
+
+### Manifest (current)
+The manifest.xml file is generated by the compiler when it compiles a module.
+This file replicates most of the information present in the binary module file,
+but in a way the loading environment can access directly without having to load
+the module.
+
+The outline of the manifest file format is as follows:
+
+    <package version="1.0">
+        <label>Human Readable Foo</label>
+        <author>Mr Magoo</author>
+        <description>Foo is a super amazing widget that will do everything for you.</description>
+        <license>commercial|dual|community</license>
+        <name>com.livecode.foo</name>
+        <version>X.Y.Z</version>
+        <type>widget|library</type>
+        <metadata key="mykey">myvalue</metadata>
+        <requires name="com.livecode.bar" version="X.Y.Z" />
+        <requires name="com.livecode.baz" version="X.Y.Z" />
+        <property name="foo" get="optional(integer)" set="optional(integer)" />
+        <property name="bar" get="string" />
+        <event name="click" parameters="in(integer),out(real)" return="optional(any)" />
+        <handler name="magic" parameters="in(integer),inout(string)" return="undefined" />
+    </package>
+
+This format works where a package contains a single module, and that module is a
+library or widget. However, it is not sufficient to describe a package which
+exports multiple widgets or library exports.
+
+### Manifest (revised)
+
+TBD
+
+### Modules (revised)
+
+The modules folder contains all the compiled module code which the package contains.
+The filenames of each module file must match the name of module contained within it
+with a .lcb suffix.
+
+A package is considered invalid if the package name is not a common strict prefix of
+all the modules it contains.
+

```

Comment:
So this is a restriction on the archive filename.  Will the name of the package also be contained within the package?  My concern is that it is all too easy for a filename to become mangled, which would then render the whole package invalid.